code/iaas/lb-logic/src/main/java/io/cattle/iaas/lb/api/action/LoadBalancerAddHostActionHandler.java

```diff
@@ -38,17 +31,8 @@ public Object perform(String name, Object obj, ApiRequest request) {
         LoadBalancer lb = (LoadBalancer) obj;
         long hostId = DataAccessor.fromMap(request.getRequestObject()).withKey(LoadBalancerConstants.FIELD_LB_HOST_ID)
                 .as(Long.class);
-        createLbHostMap(lb.getId(), hostId);
+        lbService.addHostToLoadBalancer(lb, hostId);
 
         return objectManager.reload(lb);
```

Comment:
So is it the state field for the LB that changes or some other field?  Would the trigger be that quick since the create process for the mapping is scheduled rather than executed immediately.